tests/kconfig_features/Makefile

```diff
@@ -0,0 +1,12 @@
+include ../Makefile.tests_common
+
+BOARD_WHITELIST += samr21-xpro
+
+kconfig-features:
+	@bash -c 'diff <($(MAKE) info-features-provided) \
+	    <($(MAKE) dependency-debug-features-provided-kconfig) || \
+	    (echo "ERROR: Kconfig features mismatch" && exit 1)'
+
+all: kconfig-features
+
+include $(RIOTBASE)/Makefile.include
```

Comment:
I think this test should be extended to cover other variables in `makefiles/dependencies_debug.inc.mk` that intrinsically related to provided variables, so from what is missing `BOARD CPU CPU_MODEL CPU_ARCH CPU_FAM`.